compiler/rustc_type_ir/src/elaborate.rs

```diff
@@ -132,12 +168,14 @@ impl<I: Interner, O: Elaboratable<I>> Elaborator<I, O> {
                 // Get predicates implied by the trait, or only super predicates if we only care about self predicates.
                 match self.mode {
                     Filter::All => self.extend_deduped(
+                        Some(clause),
                         cx.explicit_implied_predicates_of(data.def_id())
```

Comment:
why not instead do this filter in `implied_predicates_of` and `super_predicates_of`? :thinking: